src/Microsoft.AspNet.Mvc.ModelBinding/Binders/CompositeModelBinder.cs

```diff
@@ -34,17 +35,35 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
 
         public virtual async Task<ModelBindingResult> BindModelAsync([NotNull] ModelBindingContext bindingContext)
         {
-            var newBindingContext = CreateNewBindingContext(bindingContext,
-                                                            bindingContext.ModelName);
-
+            var newBindingContext = CreateNewBindingContext(bindingContext, bindingContext.ModelName);
             var modelBindingResult = await TryBind(newBindingContext);
-            if (modelBindingResult == null && !string.IsNullOrEmpty(bindingContext.ModelName)
-                && bindingContext.FallbackToEmptyPrefix)
+
+            if (bindingContext.FallbackToEmptyPrefix && !string.IsNullOrEmpty(bindingContext.ModelName))
             {
-                // fallback to empty prefix?
-                newBindingContext = CreateNewBindingContext(bindingContext,
-                                                            modelName: string.Empty);
-                modelBindingResult = await TryBind(newBindingContext);
+                var fallback = modelBindingResult == null;
+                if (!fallback && modelBindingResult.IsModelSet && modelBindingResult.IsEmptyModel)
```

Comment:
in the other fallback scenarios, the `ModelState` won't contain anything for the previous prefix